erpnext/assets/doctype/asset/depreciation.py

```diff
@@ -51,20 +54,35 @@ def make_depreciation_entry(asset_name, date=None):
 			je.finance_book = d.finance_book
 			je.remark = "Depreciation Entry against {0} worth {1}".format(asset_name, d.depreciation_amount)
 
-			je.append("accounts", {
+			credit_entry = {
 				"account": accumulated_depreciation_account,
 				"credit_in_account_currency": d.depreciation_amount,
 				"reference_type": "Asset",
 				"reference_name": asset.name
-			})
+			}
 
-			je.append("accounts", {
+			debit_entry = {
 				"account": depreciation_expense_account,
 				"debit_in_account_currency": d.depreciation_amount,
 				"reference_type": "Asset",
 				"reference_name": asset.name,
 				"cost_center": depreciation_cost_center
-			})
+			}
+
+			for dimension in accounting_dimensions:
+				if (asset.get(dimension['fieldname']) or dimension.get('mandatory_for_pl') or
+					dimension.get('mandatory_for_bs')):
+					credit_entry.update({
+						dimension['fieldname']: asset.get(dimension['fieldname']) or dimension.get('default_dimension')
+					})
+
+					debit_entry.update({
```

Comment:
debit_entry will be updated if `asset.get(dimension['fieldname']) or dimension.get('mandatory_for_pl')`